src/main/scala/org/deepdive/inference/datastore/SQLInferenceDataStore.scala

```diff
@@ -345,88 +278,15 @@ trait SQLInferenceDataStore extends InferenceDataStore with Logging {
-     
-    CREATE OR REPLACE FUNCTION updateid(startid bigint, sid int, sids int[], base_ids bigint[], base_ids_noagg bigint[]) RETURNS bigint AS 
-    $$
-    if '__count_1' in SD:
-      a = SD['__count_2']
-      b = SD['__count_1']
-      SD['__count_2'] = SD['__count_2'] - 1
-      if SD['__count_2'] < 0:
-        SD.pop('__count_1')
-      return startid+b-a
-    else:
-      for i in range(0, len(sids)):
-        if sids[i] == sid:
-          SD['__count_1'] = base_ids[i] - 1
-          SD['__count_2'] = base_ids_noagg[i] - 1
-      a = SD['__count_2']
-      b = SD['__count_1']
-      SD['__count_2'] = SD['__count_2'] - 1
-      if SD['__count_2'] < 0:
-        SD.pop('__count_1')
-      return startid+b-a
-      
-    $$ LANGUAGE plpythonu;
-     
-    CREATE OR REPLACE FUNCTION fast_seqassign(tname character varying, startid bigint) RETURNS TEXT AS 
-    $$
-    BEGIN
-      EXECUTE 'drop table if exists tmp_gpsid_count cascade;';
-      EXECUTE 'drop table if exists tmp_gpsid_count_noagg cascade;';
-      EXECUTE 'create table tmp_gpsid_count as select gp_segment_id as sid, count(clear_count_1(gp_segment_id)) as base_id from ' || quote_ident(tname) || ' group by gp_segment_id order by sid distributed by (sid);';
-      EXECUTE 'create table tmp_gpsid_count_noagg as select * from tmp_gpsid_count distributed by (sid);';
-      EXECUTE 'update tmp_gpsid_count as t set base_id = (SELECT SUM(base_id) FROM tmp_gpsid_count as t2 WHERE t2.sid <= t.sid);';
-      RAISE NOTICE 'EXECUTING _fast_seqassign()...';
-      EXECUTE 'select * from _fast_seqassign(''' || quote_ident(tname) || ''', ' || startid || ');';
-      RETURN '';
-    END;
-    $$ LANGUAGE 'plpgsql';
-     
-    CREATE OR REPLACE FUNCTION _fast_seqassign(tname character varying, startid bigint)
-    RETURNS TEXT AS
-    $$
-    DECLARE
-      sids int[] :=  ARRAY(SELECT sid FROM tmp_gpsid_count ORDER BY sid);
-      base_ids bigint[] :=  ARRAY(SELECT base_id FROM tmp_gpsid_count ORDER BY sid);
-      base_ids_noagg bigint[] :=  ARRAY(SELECT base_id FROM tmp_gpsid_count_noagg ORDER BY sid);
-      tsids text;
-      tbase_ids text;
-      tbase_ids_noagg text;
-    BEGIN
-      SELECT INTO tsids array_to_string(sids, ',');
-      SELECT INTO tbase_ids array_to_string(base_ids, ',');
-      SELECT INTO tbase_ids_noagg array_to_string(base_ids_noagg, ',');
-      if ('update ' || tname || ' set id = updateid(' || startid || ', gp_segment_id, ARRAY[' || tsids || '], ARRAY[' || tbase_ids || '], ARRAY[' || tbase_ids_noagg || ']);')::text is not null then
-        EXECUTE 'update ' || tname || ' set id = updateid(' || startid || ', gp_segment_id, ARRAY[' || tsids || '], ARRAY[' || tbase_ids || '], ARRAY[' || tbase_ids_noagg || ']);';
-      end if;
-      RETURN '';
-    END;
-    $$
-    LANGUAGE 'plpgsql';
-    """
-
   def init() : Unit = {
   }
 
+  // given a path, get file/folder name
+  // e.g., /some/path/to/folder -> folder
+  def getFileNameFromPath(path: String) : String = {
```

Comment:
How is calling "getFileNameFromPath(p)" better than "new File(p).getName()"?  Just to be all Scala-like?